main.tf

```diff
@@ -206,13 +206,58 @@ data "aws_caller_identity" "default" {
 data "aws_region" "default" {
 }
 
+locals {
+  image_tag      = var.image_tag != "" ? var.image_tag : "$(echo $CODEBUILD_RESOLVED_SOURCE_VERSION | cut -c 1-7)"
+  has_taskdef    = var.codedeploy_taskdefinition != ""
+  has_appspec    = var.codedeploy_appspec != ""
+  taskdef_writer = <<EOF
+      - echo Writing task definition file...
+      - echo "${var.codedeploy_taskdefinition}" > ${var.codedeploy_taskdefinition_path}
+  EOF
+  appspec_writer = <<EOF
+      - echo Writing app spec file...
+      - echo "${var.codedeploy_appspec}" > ${var.codedeploy_appspec_path}
+  EOF
+
+  buildspec = var.buildspec != "" ? var.buildspec : <<EOF
```

Comment:
Instead of a heredoc, a separate `.tmpl` or `.tpl` file and the use of the `templatefile` function would be better.